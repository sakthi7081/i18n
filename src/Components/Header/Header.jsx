import React from 'react';
import { StyledPaper,StyledBox,HeaderIcon } from './Header.style';

const Header = () => {
    return (
        <StyledPaper elevation={0} data-test='Header-Component'>
            <StyledBox data-test='Title'>
            <HeaderIcon/>
                Multi Language
            </StyledBox>
            
        </StyledPaper>        
    );
};

export default Header;