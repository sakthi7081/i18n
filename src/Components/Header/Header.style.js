import Paper from "@material-ui/core/Paper";
import styled from "styled-components";
import TranslateIcon from "@material-ui/icons/Translate";
import { color } from "../../Const";

export const StyledPaper = styled(Paper)`
  &&& {
    background-color: ${color};
    overflow: hidden;
    margin-bottom: 10px;
    height: 50px;
    align-items: center;
    display: flex;
    color: #fff;
    border-radius: 0;
  }
`;

export const StyledBox = styled("div")({
  display: "flex",
  marginLeft: "20px",
  justifyContent: "center",
  fontWeight: "bold",
  fontSize: "20px",
});

export const HeaderIcon = styled(TranslateIcon)({
  marginRight: "10px",
});
