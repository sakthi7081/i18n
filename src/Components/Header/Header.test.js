import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { findTestAttr } from "./../../Utility/Utility";
import Header from "./Header";

const setUp = () => {
  const component = shallow(<Header />);
  return component;
};

describe("Header Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render the component with no error", () => {
    const wrapper = findTestAttr(component, "Header-Component");
    expect(wrapper.length).toBe(1);
  });

  it("Should render the Title", () => {
    const wrapper = findTestAttr(component, "Title");
    expect(wrapper.length).toBe(1);
  });

  it("Should match the snapShot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
