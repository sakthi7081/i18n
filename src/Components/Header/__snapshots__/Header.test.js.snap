// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should match the snapShot 1`] = `
<Styled(WithStyles(ForwardRef(Paper)))
  data-test="Header-Component"
  elevation={0}
>
  <styled.div
    data-test="Title"
  >
    <Styled(TranslateIcon) />
    Multi Language
  </styled.div>
</Styled(WithStyles(ForwardRef(Paper)))>
`;
